lib/common/bits.h

```diff
@@ -34,16 +37,11 @@ MEM_STATIC unsigned ZSTD_highbit32(U32 val)   /* compress, dictBuilder, decodeCo
 #       if STATIC_BMI2 == 1
             return _lzcnt_u32(val)^31;
 #       else
-            if (val != 0) {
-                unsigned long r;
-                _BitScanReverse(&r, val);
-                return (unsigned)r;
-            } else {
-                /* Should not reach this code path */
-                __assume(0);
-            }
+            unsigned long r;
```

Comment:
Note: I removed `if (val != 0)` for consistency with the GNUC branch (there is an `assert(val != 0)` at the top of the function). If this is unsafe for some reason (we are worried about security when the assert is compiled out or something), please let me know and I'll add a commit reverting the change.

Reply:
While I believe the `assert()` is correct for this situation,
I also suspect that the `if (val != 0)` was added by a 3rd party contributor
because without it, some windows static analyzer would complain in production mode (essentially assuming the `val == 0` is a possible scenario).
Mentioning it because the issue might come back after this change.

Reply:
@Cyan4973  Makes sense, I'll revert the change. It only affects Win32 anyway.

Reply:
I think for this PR it makes sense to keep the current implementation (just to keep the scope of this PR manageable). I think that since this only affects Win32, and the performance cost is just a single 100%-predictable branch, it may not be worth the time to figure out a less expensive way to appease the static analyzers. But if someone wants to look into this it would definitely be welcome.